### PR TITLE
feat(k3s): add knob for configuring k3s kubelet

### PR DESCRIPTION
**Pull request by stormqueen1990**
Include an option for specifying an inline KubeletConfig manifest that will get applied to the underlying k3s cluster.

### CODE REVIEW

File: internal/harnesses/k3s/k3s.go [hunk 1]
```diff
@@ -121,13 +122,25 @@ func New(id string, cli *provider.DockerClient, opts ...Option) (types.Harness,
 		}
 	}
 
+	extraFiles := make([]provider.File, 0)
+	if "" != k3s.opt.KubeletConfig {
+		kubeletConfigBuf := &bytes.Buffer{}
+		kubeletConfigBuf.WriteString(k3s.opt.KubeletConfig)
```
Comment (joshrwolf) [inline internal/harnesses/k3s/k3s.go:128]: if we type `k3s.opt.KubeletConfig` then we can just marshal to json here

File: internal/provider/harness_k3s_resource.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ type HarnessK3sResourceModel struct {
 	Timeouts             timeouts.Value                           `tfsdk:"timeouts"`
 	Resources            *ContainerResources                      `tfsdk:"resources"`
 	Hooks                *HarnessHooksModel                       `tfsdk:"hooks"`
+	KubeletConfig        types.String                             `tfsdk:"kubelet_config"`
```
Comment (joshrwolf) [inline internal/provider/harness_k3s_resource.go:63]: this is fine staying as a string that we probably validate can be marshal'd. I don't know of an easy way to `types.Any` or `types.Json` in tf

File: internal/harnesses/k3s/k3s.go (hunk 1 above)
Comment (stormqueen1990) [reply internal/harnesses/k3s/k3s.go:128]: `Contents` expects an `io.Reader`, but `[]byte` isn't an `io.Reader`. What did you have in mind here?

Comment (stormqueen1990) [reply internal/harnesses/k3s/k3s.go:128]: I don't know that this is a good idea. Marshalling to an object then unmarshalling causes a lot more fields to be written out to the final configuration, and that will end up resulting in non-specified fields showing up in the configuration.

File: internal/provider/harness_k3s_resource.go [hunk 3]
```diff
@@ -326,6 +341,23 @@ You can access the cluster with something like: "KUBECONFIG=%s kubectl get po -A
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
+func validateKubeletConfig(content string) error {
```
Comment (joshrwolf) [inline internal/provider/harness_k3s_resource.go:344]: I like this validation approach but can we move it closer to its usage (in the `k3s`) package?

probably just copy/pasta the function into a k3s `WithKubeletConfig` `Option`

Comment (stormqueen1990) [reply internal/provider/harness_k3s_resource.go:344]: Sure!

Comment (stormqueen1990) [reply internal/provider/harness_k3s_resource.go]: Updated!

**Review by joshrwolf — APPROVED** [review]

**Pull request merged**